xml/src/main/java/org/ehcache/xml/ConfigurationParser.java

```diff
@@ -105,6 +112,23 @@
   private final Map<URI, CacheResourceConfigurationParser> resourceXmlParsers = new HashMap<URI, CacheResourceConfigurationParser>();
   private final ConfigType config;
 
+  static String getReplacementString(String originalValue) {
+    Matcher matcher = SYSPROP.matcher(originalValue);
+
+    StringBuffer sb = new StringBuffer();
+    while (matcher.find()) {
+      final String property = matcher.group(1);
+      final String value = System.getProperty(property);
+      if (value == null) {
+        return null;
+      }
+      matcher.appendReplacement(sb, value);
```

Comment:
That value is not used as a literal string, unescaped `$1` or `${foo}` sequences will be replaced with the relevant capture group from the match.  You could call this a feature, but I don't think it's one we want.  `Matcher.quoteReplacement(value)` quotes the string properly so that it behaves like a literal replacement.